### PR TITLE
Declaration of item recipes.

### PR DESCRIPTION
**Pull request by crojewsk**
Following have been included in this patch:
- troll recipes
- armory recipes
- workshop recipes
- witch doctor hut recipes
- tannery recipes
- mixing pot recipes

Initialization block is not included i.e. no recipe initialization is
actually invoked - it will be a part of patch responsible for removal
of old, deprecated item recipes.

### CODE REVIEW

File: wurst/lib/ItemRecipes.wurst [hunk 1]
```diff
@@ -0,0 +1,1431 @@
...
+        ..addItem(ITEM_STICK)
+        ..addItem(ITEM_STICK)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_LOADED_THISTLES, 15, true, true, true)
+        ..addItem(ITEM_BLOW_GUN)
+        ..addItem(ITEM_THISTLES)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_SALVE_ARMOR, 1, true, false, true)
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: These shouldn't be disassembled

File: wurst/lib/ItemRecipes.wurst [hunk 2]
```diff
@@ -0,0 +1,1431 @@
...
+
+    recipe = new ItemRecipe(ITEM_LOADED_THISTLES, 15, true, true, true)
+        ..addItem(ITEM_BLOW_GUN)
+        ..addItem(ITEM_THISTLES)
+    recipe.requirement = requirement
+        // add requirement for troll unit to
+
+    recipe = new ItemRecipe(ITEM_LOADED_DARK_THISTLES, 12, true, true, true)
+        ..addItem(ITEM_BLOW_GUN)
+        ..addItem(ITEM_DARK_THISTLES)
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: What happens with the other three?

Comment (theQuazz) [reply wurst/lib/ItemRecipes.wurst]: Is this sensitive to the number of charges that are in the `ITEM_DARK_THISTLES`?

File: wurst/lib/ItemRecipes.wurst [hunk 3]
```diff
@@ -0,0 +1,1431 @@
...
+        ..addItem(ITEM_STICK)
+        ..addItem(ITEM_CLAY_BALL)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_DEFENSIVE_WARD_KIT, 1, true, true, true)
+        ..addItem(ITEM_STONE)
+        ..addItem(ITEM_STONE)
+        ..addItem(ITEM_MANA_CRYSTAL)
+        ..addItem(ITEM_IRON_INGOT, false, 0)
+    recipe.requirement = requirement
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: You can delete this one

File: wurst/lib/ItemRecipes.wurst [hunk 4]
```diff
@@ -0,0 +1,1431 @@
...
+        ..addItem(ITEM_BATTLE_SHIELD)
+        ..addItem(ITEM_SPIRIT_WIND)
+        ..addItem(ITEM_SPIRIT_WATER)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_NAPALM, 1, true, false, true)
+        ..addItem(ITEM_FIRE_BOMB)
+        ..addItem(ITEM_FIRE_BOMB)
+        ..addItem(ITEM_FLINT)
+    recipe.requirement = requirement
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: Delete this one

File: wurst/lib/ItemRecipes.wurst [hunk 5]
```diff
@@ -0,0 +1,1431 @@
...
+
+    recipe = new ItemRecipe(ITEM_BATTLE_SUIT, 0, true, false, true)
+        ..addItem(ITEM_BATTLE_AXE)
+        ..addItem(ITEM_BATTLE_ARMOR)
+        ..addItem(ITEM_BATTLE_GLOVES)
+        ..addItem(ITEM_BATTLE_SHIELD)
+        ..addItem(ITEM_SPIRIT_WIND)
+        ..addItem(ITEM_SPIRIT_WATER)
+    recipe.requirement = requirement
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: Delete this one

File: wurst/lib/ItemRecipes.wurst [hunk 6]
```diff
@@ -0,0 +1,1431 @@
...
+
+    recipe = new ItemRecipe(ITEM_SMOKE_BOMB, 3, true, false, true)
+        ..addItem(ITEM_ACORN)
+        ..addItem(ITEM_SPIRIT_WIND)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_DARK_THISTLES, 3, true, false, true)
+        ..addItem(ITEM_THISTLES)
+        ..addItem(ITEM_DARK_ROCK)
+    recipe.requirement = requirement
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: What if `ITEM_THISTLES` has 15 charges?

File: wurst/lib/ItemRecipes.wurst [hunk 7]
```diff
@@ -0,0 +1,1431 @@
...
+        ..addItem(ITEM_BONE_SHIELD)
+        ..addItem(ITEM_ELK_HIDE)
+        ..addItem(ITEM_STEEL_INGOT)
+        ..addItem(ITEM_SPIRIT_WIND)
+        ..addItem(ITEM_SPIRIT_WATER)
+        ..addItem(ITEM_MANA_CRYSTAL)
+    recipe.requirement = requirement
+
+    recipe = new ItemRecipe(ITEM_FIRE_BOMB, 2, true, false, true)
+        ..addItem(ITEM_SMOKE_BOMB)
```
Comment (theQuazz) [inline wurst/lib/ItemRecipes.wurst]: What if `SMOKE_BOMB` has 1 charge?

Comment (Aluvian-Sprinter) [reply wurst/lib/ItemRecipes.wurst]: The current recipe does not take into account smoke bomb charges either, it produces 2 fire bombs regardless.

Comment (crojewsk) [reply wurst/lib/ItemRecipes.wurst]: Precisely. For the rest of conflicts, I'll be planning ItemRecipe update for today to fix them.

File: wurst/lib/ItemRecipes.wurst (hunk 6 above)
Comment (crojewsk) [reply wurst/lib/ItemRecipes.wurst]: Now properly adjusts charges based on ingredients used.

**Pull request merged**